build/tf-wrapper.sh

```diff
@@ -23,78 +23,189 @@ project_id=$4
+
+#====================================================================#
+# Function to replace '/' with '-' to convert a path to a file name
+#====================================================================#
+convert_path() {
+  echo "$1" | sed -r 's/\//-/g'
+}
 
 ## Terraform apply for single environment.
 tf_apply() {
   local path=$1
-  local tf_env=$2
-  local tf_component=$3
+  local tf_env="${path#$base_dir/}"
+  local tf_file
+  tf_file="$(convert_path "$tf_env")"
   echo "*************** TERRAFORM APPLY *******************"
-  echo "      At environment: ${tf_component}/${tf_env} "
+  echo "      At environment: ${tf_env} "
   echo "***************************************************"
   if [ -d "$path" ]; then
     cd "$path" || exit
-    terraform apply -input=false -auto-approve "${tmp_plan}/${tf_component}-${tf_env}.tfplan" || exit 1
+    terraform apply -input=false -auto-approve "${tmp_plan}/${tf_file}.tfplan" || exit 1
     cd "$base_dir" || exit
   else
-    echo "ERROR:  ${path} does not exist"
+    echo "ERROR: ${path} does not exist"
   fi
 }
 
 ## terraform init for single environment.
 tf_init() {
   local path=$1
-  local tf_env=$2
-  local tf_component=$3
+  local tf_env="${path#$base_dir/}"
   echo "*************** TERRAFORM INIT *******************"
-  echo "      At environment: ${tf_component}/${tf_env} "
+  echo "      At environment: ${tf_env} "
   echo "**************************************************"
   if [ -d "$path" ]; then
     cd "$path" || exit
     terraform init || exit 11
     cd "$base_dir" || exit
   else
-    echo "ERROR:  ${path} does not exist"
+    echo "ERROR: ${path} does not exist"
   fi
 }
 
 ## terraform plan for single environment.
 tf_plan() {
   local path=$1
-  local tf_env=$2
-  local tf_component=$3
+  local tf_env="${path#$base_dir/}"
+  local tf_file
+  tf_file="$(convert_path "$tf_env")"
   echo "*************** TERRAFORM PLAN *******************"
-  echo "      At environment: ${tf_component}/${tf_env} "
+  echo "      At environment: ${tf_env} "
   echo "**************************************************"
   if [ ! -d "${tmp_plan}" ]; then
     mkdir "${tmp_plan}" || exit
   fi
   if [ -d "$path" ]; then
     cd "$path" || exit
-    terraform plan -input=false -out "${tmp_plan}/${tf_component}-${tf_env}.tfplan" || exit 21
+    terraform plan -input=false -out "${tmp_plan}/${tf_file}.tfplan" || exit 21
     cd "$base_dir" || exit
   else
-    echo "ERROR:  ${tf_env} does not exist"
+    echo "ERROR: ${path} does not exist"
   fi
 }
 
-## terraform init/plan/validate for all valid environments matching regex.
+#============================================================================#
+# terraform init/plan/validate for all valid environments matching condition.
+#============================================================================#
 tf_plan_validate_all() {
-  local env
-  local component
+  local leaf
   find "$base_dir" -mindepth 1 -maxdepth 1 -type d \
```

Comment:
Should the mindepth & maxdepth be set to variables too?

Reply:
Hi @jasonbisson , thanks for the review

The sole purpose of this code section and the other code section is to scan the base directory of the repository to filter directories that do not contain terraform configuration that should be considered for evaluation.

The result of this filter is the initial set of directories we would use as our base scan

For this code section, "mindepth" should always be one because we don’t want to miss any directory and "maxdepth" should always be one too so that the filtered list will always have the same depth.  if we had for example  "maxdepth" 2 we would have in the result list

- envs
- envs/shared

as starting point for the next step

This is an initial filter to trim directories that we know should not contain configurations for the environments.